CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ include_directories(SYSTEM
   ${OPENGL_INCLUDE_DIR}
   ${PYTHON_INCLUDE_PATH}
 )
-include_directories(src ${catkin_INCLUDE_DIRS})
+include_directories(src SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
Could you elaborate on that? I have never had problems with marking include directories as system directories and it is the cleanest way for the compiler to ignore errors in external headers. It is the least compiler specific and it does not depend on the exact contents of the external headers, which is not under control of this package (so can change and fix or add new warnings).